src/linker/Linker/SerializationHelper.cs

```diff
@@ -0,0 +1,123 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Mono.Cecil;
+using Mono.Linker.Dataflow;
+using Mono.Linker.Steps;
+
+namespace Mono.Linker
+{
+
+	// This recursively marks members of types for serialization. This is not supposed to be complete; it is just
+	// here for back-compat with the old behavior of xamarin-android.
+	//
+	// The xamarin-android behavior was as follows:
+	//
+	// Discover members in the "link" assemblies with certain attributes:
+	//   for XMLSerializer: Xml*Attribute, except XmlIgnoreAttribute
+	//   for DataContractSerializer: DataContractAttribute or DataMemberAttribute
+	// These members are considered "roots" for serialization.
+	//
+	// For each "root":
+	//    in an SDK assembly, set TypePreserve.All for types, or conditionally preserve methods or property methods
+	//      event methods were not preserved.
+	//    in a non-SDK assembly, mark types, fields, methods, property methods, and event methods
+	//    recursively scan types of properties and fields (including generic arguments)
+	//      for each recursive type, conditionally preserve the default ctor
+	//
+	// We want to match the above behavior in a more consistent way (erring on the side of being more conservative).
+
+	// Instead of conditionally preserving things, we will just mark them, and we will do so consistently for every
+	// type discovered as part of the type graph reachable from the discovered roots. We also do not distinguish between
+	// SDK and non-SDK assemblies.
+	//
+	// The behavior is as follows:
+	//
+	// Discover attributed "roots" the same way.
+	//
+	// For each "root":
+	//   recursively scan types of properties and fields (including generic arguments)
+	//     for each recursive type:
+	//       mark the type, and set TypePreserve.All
+	//         this handles the cases where XA used to set TypePreserve.All, preserve default ctors, and mark or conditionally preserve fields/methods (including property and event methods)
+
+	public class SerializationHelper
```

Comment:
You know I hate Helper names ;-). What about SerializationMarker instead ?

Reply:
Good idea, fixed :)